### PR TITLE
Using Vault to store secrets

### PR DESCRIPTION
**Pull request by fgimenez**
The changes for the sync script come in a separate PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,20 @@ In order to deploy a new server it may be useful to keep the configuration from
...
 
+## Secrets management
+
+We use Hashicorp's [https://www.vaultproject.io/](Vault) to manage the secrets used by the jenkins deployment. There is a script for setting up this service on a cloud environment, you can provision it with:
+
+    $ ./bin/secrets/cloud-provision.sh <private_key_path> <openstack_credentials_path>
+
+being `<private_key_path>` the ssh key that the slaves will use for creating images and accessing instances created from them, and `<openstack_credentials_path>` the path of the novarc file needed for accessing the cloud resources where the testbeds will be created. These secrets will be created under the vault path `secrets/jenkins/tests/ssh-key` and `secrets/jenkins/tests/openstack-credentials` respectively.
+
+After the provisioning a `vault-remote.txt` text file is created in the current directory containing the server keys and root token, keep it safe because they are unique for each deployment.
```
Comment (elopio) [inline README.md]: After the provisioning, a `vault...
(you are missing a comma)

File: config/vault/cluster.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      - 8600:53/udp
+      - 8400:8400
+      - 8500:8500
+    command: "-server -bootstrap -ui-dir /ui"
+    volumes:
+      - /mnt:/data
+    restart: always
+
+  vault:
+    image: fgimenez/vault
```
Comment (elopio) [inline config/vault/cluster.yml]: why don't we put it in the team right away?

**Comment by elopio** [comment]
This looks awesome Federico!
I left a couple of ignorable tiny comments because I feel bad when I have nothing to say :blush:

**Comment by fgimenez** [comment]
Done, thanks! I've also taken the opportunity to add the consul and vault containers to travis' checks.

**Pull request merged**